libsignal-protocol/src/store_context.rs

```diff
@@ -106,6 +106,29 @@ impl StoreContext {
         }
     }
 
+    /// Load the sub-device sessions corresponding to the provided recipient
+    /// identifier.
+    pub fn get_sub_device_sessions(
+        &self,
+        identifier: &str,
+    ) -> Result<Vec<i32>, Error> {
+        unsafe {
+            let mut sessions = ptr::null_mut();
+            sys::signal_protocol_session_get_sub_device_sessions(
+                self.raw(),
+                &mut sessions,
+                identifier.as_ptr() as *const i8,
+                identifier.len(),
+            )
+            .into_result()?;
+            let mut ids = vec![];
```

Comment:
Maybe init with `with_capacity(sys::signal_int_list_size(sessions))`?

```suggestion
            let mut ids = Vec::with_capacity(sys::signal_int_list_size(sessions));
```